src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2019 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import {useEffect, useRef, useState} from "react";
+
+// Hook behaving like useState but persisting the value to localStorage. Returns same as useState
+export const useLocalStorageState = (key: string, initialValue: boolean) => {
+    const lsKey = useRef("useLocalStorageState_" + key).current;
```

Comment:
Why do you need the `useRef` hook here? This is to track changes to the key...?

About the item name that's used in local storage, adding a prefix of `useLocalStorageState_` seems to bake in an implementation detail into the data being stored. Maybe use a generic `mx_` prefix, or just leave it up to the caller?

Reply:
The hook here prevents the value from ever changing, it takes the initial value and stores it in the ref

> seems to bake in an implementation detail into the data being stored.

given that it serializes + deserializes it is implementation specific, no?

Reply:
Instead I could make it save the old key and load the value form the new key on a change, that would be more re-usable

Reply:
> given that it serializes + deserializes it is implementation specific, no?

Hmm, I don't know if that part is really that specific to exactly this function, it's fairly common to do that with local storage. 

The main thing I am reacting to is the name `useLocalStorageState` has little / nothing to do with the data being stored, so it's a bit surprising to embed that in key name being stored. Maybe later we'll rewrite this code into a function with a different name, but we'd still have a bunch of keys stored in local storage with the prefix `useLocalStorageState_`. The name of a specific data access function that happens to exist at the moment seems like too specific and too temporary a thing to place in a storage layer to me.